ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFDateFormat.java

```diff
@@ -106,25 +102,22 @@ public Object evaluate(DeferredObject[] arguments) throws HiveException {
       return null;
     }
 
-    ZoneId id = (SessionState.get() == null) ? new HiveConf().getLocalTimeZone() : SessionState.get().getConf()
-        .getLocalTimeZone();
     // the function should support both short date and full timestamp format
     // time part of the timestamp should not be skipped
     Timestamp ts = getTimestampValue(arguments, 0, tsConverters);
+
     if (ts == null) {
-      Date d = getDateValue(arguments, 0, dtInputTypes, dtConverters);
-      if (d == null) {
-        return null;
-      }
-      ts = Timestamp.ofEpochMilli(d.toEpochMilli(id), id);
+      return null;
     }
 
+    // We are doing an extra timestamp conversion from timeZone to UTC and then UTC to timeZone to maintain the
+    // same zone format semantics as that of SimpleDateFormat for some specific locale. E.g. PST/PDT, CST
+    // If this conversion is not there PDT -> PT and CST -> CT
```

Comment:
Reuse "ts" itself instead of new ts2 object.

Reply:
removed ts2